### ISSUE TITLE
Bug: jsid_cache breaks for `"toString"`, and I suspect other methods of JS Object

### ISSUE THREAD

**Issue by michaeljb** [opal/opal#2583]
**Describe the bug**

Opal version: v1.7.0 - v1.7.3

The cache added in #2481 breaks for the method `toString`, and I suspect will break for other Object methods, like `valueOf`.

`jsid_cache["toString"]` evaluates to the `toString()` method on the cache object itself, instead of resulting in a cache miss and building the string `"$toString"`.

https://github.com/opal/opal/blob/f02fa40875c51b388de9c1b534459d363bcf951e/opal/corelib/runtime.js#L1868

**To Reproduce**

Call `Opal.add_stubs('toString')`, or write some Ruby code that will compile with such a call.

This was discovered while trying to upgrade [18xx.games](https://18xx.games/) to use Opal 1.7. We have a class which wraps the JS [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) class, and it needs to call `toString` on the wrapped object, leading to a runtime error:

https://github.com/tobymao/18xx/blob/49a570c9237ac9739023db780f69a58c336fcd1b/assets/app/lib/params.rb#L36-L38

**Comment by hmdne** [comment]
```ruby
      def to_query_string
        @native&.toString() || ''
      end
```

Try this, before we release the fix:

```ruby
      def to_query_string
        `Opal.truthy(self.native) ? self.native.$toString() : nil` || ''
      end
```

**Comment by michaeljb** [comment]
Looks like that gives the same error.

```
Uncaught TypeError: self.native.$toString is not a function
    $$to_query_string http://localhost:9292/assets/main.js:36171
    $$render_header http://localhost:9292/assets/main.js:57956
    $$render http://localhost:9292/assets/main.js:57937
    $$h http://localhost:9292/assets/deps.js:27599
    $$render_row http://localhost:9292/assets/main.js:58203
    $$render http://localhost:9292/assets/main.js:58140
    $$h http://localhost:9292/assets/deps.js:27599
    $$render_content http://localhost:9292/assets/main.js:35335
    $$render http://localhost:9292/assets/main.js:35323
    $Component_update$excl$2 http://localhost:9292/assets/deps.js:27627
    $$attach http://localhost:9292/assets/deps.js:27544
    <anonymous> http://localhost:9292/:14
main.js:36171:75
```

Here's the compiled JS from the debugger tab:

```
        return $def(self, '$to_query_string', function $$to_query_string() {
          var self = this, $ret_or_1 = nil;

          if ($truthy(($ret_or_1 = Opal.truthy(self.native) ? self.native.$toString() : nil))) {
            return $ret_or_1
          } else {
            return ""
          }
        });
```

**Comment by hmdne** [comment]
Makes sense. Try this then:

```ruby
      def to_query_string
        `Opal.truthy(self.native) ? self.native.native.toString() : nil` || ''
      end
```

**Comment by michaeljb** [comment]
Looks like that works, thank you!

```
        return $def(self, '$to_query_string', function $$to_query_string() {
          var self = this, $ret_or_1 = nil;

          if ($truthy(($ret_or_1 = Opal.truthy(self.native) ? self.native.native.toString() : nil))) {
            return $ret_or_1
          } else {
            return ""
          };
        });
```

**Comment by hmdne** [comment]
@michaeljb FYI - we have released v1.7.4 with the fix.

**Comment by michaeljb** [comment]
@hmdne awesome, thanks!! (I recently got back from a vacation so I didn’t see this in a more timely manner 😄 )
